hyperjumptech/monika
# Clean error msg when failed to load config file

# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Addresses issue #1036 
2. Cleans up error message when config failed to be loaded

## How did you implement / how did you fix it  
1.  Catch the thrown error
2. Exit the program when not in symonMode, 

## How to test  
1. Use a bad formatted config.yaml 
```yaml
probes:
  - id: 'http-1'
    name: 'status-200-test'
    requests:
      urlsdfsd: https://httpbin.org/status/404
    alerts:
      - assertion: response.status < 200 or response.status > 299
        message: HTTP Status is not 200
      - assertion: response.time > 2000
        message: Response time is more than 2000ms
    incidentThreshold: 3
    recoveryThreshold: 3

notifications:
  - id: desktop
    type: desktop
```
2. Run monika with the config `monika -c config.yaml`
3. Expect a friendlier and more usefule error message 
4. For development, `export NODE_ENV=development` to get full stack dump

## Before PR

Unhandled formate/parsing error
![image](https://github.com/hyperjumptech/monika/assets/964106/b4472631-6326-45f0-b8d5-a7a7d8a00197)

## After PR

Catches all formatting error
![image](https://github.com/hyperjumptech/monika/assets/964106/a698cba1-9504-43f2-979c-fc119c692df2)
